addons/web_editor/static/src/js/editor/odoo-editor/src/OdooEditor.js

```diff
@@ -4195,6 +4195,40 @@ export class OdooEditor extends EventTarget {
             return;
         }
         const anchorNode = selection.anchorNode;
+        const isSelectionInEditable = this.isSelectionInEditable(selection);
+        if (this._isMouseStateActive && selection.isCollapsed && isSelectionInEditable) {
+            const range = getDeepRange(this.editable);
+            const { startContainer, startOffset, endContainer, endOffset } = range;
+            const linkElement = closestElement(startContainer, "a");
+            if (linkElement) {
+                const parentBlockElement = closestBlock(linkElement);
+                const linkIndex = [...parentBlockElement.childNodes].indexOf(linkElement);
+                const linkDescendants = descendants(linkElement);
+                const isZeroWidthSpace = (node) =>
+                    node.nodeType === Node.TEXT_NODE && node.nodeValue === "\uFEFF";
+
+                // Check if the cursor is positioned at the begining of link.
+                const isCursorAtStartOfLink = isZeroWidthSpace(startContainer)
+                    ? linkDescendants.indexOf(startContainer) === 0
+                    : startContainer.nodeType === Node.TEXT_NODE &&
+                    linkDescendants.indexOf(startContainer) === 1 &&
+                    startOffset === 0;
+
+                // Check if the cursor is positioned at the end of link.
+                const isCursorAtEndOfLink = isZeroWidthSpace(endContainer)
+                    ? linkDescendants.indexOf(endContainer) === linkDescendants.length - 1
+                    : endContainer.nodeType === Node.TEXT_NODE &&
+                    linkDescendants.indexOf(endContainer) === linkDescendants.length - 2 &&
+                    endOffset === nodeSize(endContainer);
+
+                // Handle selection movement.
+                if (isCursorAtStartOfLink || isCursorAtEndOfLink) {
+                    ev.preventDefault();
+                    delete this._isMouseStateActive;
```

Comment:
This should be out of current `if` block because `_isMouseStateActive` will always be true and if the cursor is not at start or end of link or even within link this will never be deleted. Also what if you click at the same offset, `onSelectionChange` will not be called and `_isMouseStateActive` will always remain active. :(